compiler/src/dotty/tools/dotc/core/Substituters.scala

```diff
@@ -195,4 +195,28 @@ trait Substituters { this: Context =>
   final class SubstParamsMap(from: BindingType, to: List[Type]) extends DeepTypeMap {
     def apply(tp: Type): Type = substParams(tp, from, to, this)
   }
+
+  /** An approximating substitution that can handle wildcards in the `to` list */
+  final class SubstApproxMap(from: List[Symbol], to: List[Type])(implicit ctx: Context) extends ApproximatingTypeMap {
```

Comment:
The regular `SubstBindingMap` also extends `DeepTypeMap`, does this matter here ?

Reply:
No, we do not need to substitute ClassInfos here.